kahuna/public/js/components/gr-add-label/gr-add-label.html

```diff
@@ -1,9 +1,21 @@
 <button ng:class="{'small': ctrl.grSmall}"
-        ng:click="ctrl.addLabel()"
+        ng:click="ctrl.active = true"
         ng:disabled="ctrl.adding"
+        ng:if="!ctrl.active"
         title="Add label">
     <gr-icon ng:class="{'spin': ctrl.adding}">add_box</gr-icon>
-    <span ng:hide="ctrl.grSmall">
-        Add<span ng:show="ctrl.adding">ing...</span>
-    </span>
+        <span ng:hide="ctrl.grSmall">
+            Add<span ng:show="ctrl.adding">ing...</span>
+        </span>
 </button>
+<form ng:if="ctrl.active" ng:submit="ctrl.addLabel()">
+    <input type='text' ng:model="ctrl.newLabel" gr:auto-focus placeholder="label1, label2, label3..."></input>
+    <span class="gr-add-label__buttons">
+        <button class="button-cancel" type="button" ng:click="ctrl.cancel()" title="Close">
+            <gr-icon-label gr-icon="close"><span ng:hide="ctrl.grSmall">Cancel</span></gr-icon-label>
+        </button>
+        <button class="button-save" type="submit" title="Save new label">
+            <gr-icon-label gr-icon="check"><span ng:hide="ctrl.grSmall">Save</span></gr-icon-label>
```

Comment:
Instead of all the `grSmall` stuff, we could have two directives, one called `add-label` and one called `add-label-small` to avoid all this visual logic?